### DISCUSSION TITLE
WHERE filtering options?

### DISCUSSION THREAD

**Discussion by seven-of-eleven** [neilotoole/sq#434] [Q&A] [answered]
I read through some of the docs and worked through the tutorial. I noted that you can filter:
> sq '.actor | .first_name, .last_name | where(.first_name == "TOM”)’

which successfully shows the records with TOM as first_name. This seems to require an exact match. 

As example how would I search for anything that starts with “TO” or perhaps contains “TO”, to show record for TOM, TORRY, TONY, TOBIAS… I know SQL generally has a LIKE operator with % and _ options. I tried using * and % but neither worked.

So I guess my questions is if there are options to search for partial match of a string?

**Comment by seven-of-eleven** [comment]
I solved this somewhat by removing the `where` filter in sq and piping the sq results into jq. Then I used jq to filter the results.
> sq -j '.actor | .first_name, .last_name' | jq '.[] | select(.first_name | startswith("MA”))’
sq -j '.actor | .first_name, .last_name' | jq '.[] | select(.first_name | contains("MA”))’
sq -j '.actor | .first_name, .last_name' | jq '.[] | select(.first_name | endswith("T”))’

This works, but let me know if there's a way to do this within sq.

Thanks.

**Reply by neilotoole** [reply]
Hi @seven-of-eleven,

`sq` doesn't have the equivalent of `LIKE`, or the `jq` string testing operators yet. I'll put it on the todo-list!

**Reply by seven-of-eleven** [reply]
Just to follow up on this when I used sql it also did work:
> sq sql "SELECT \"First Name\",\"Last Name\",City,Email FROM data WHERE (\"Last Name\" LIKE 'B%' AND \"First Name\" LIKE 'S%') ORDER BY \"Last Name\"" -C

Incase someone else runs into this issue. 
Best

**Comment by neilotoole** [comment, marked as answer]
@seven-of-eleven: as of [**v0.53.0**](https://github.com/neilotoole/sq/releases/tag/v0.53.0) (2026-05-25), SLQ has native partial-string matching, so you no longer need to pipe through `jq`.

Eight new functions cover the common cases:

**Literal substring (case-sensitive):**

- [`contains`](https://sq.io/docs/query#contains)
- [`startswith`](https://sq.io/docs/query#startswith)
- [`endswith`](https://sq.io/docs/query#endswith)

**Case-insensitive variants:**

- [`icontains`](https://sq.io/docs/query#icontains)
- [`istartswith`](https://sq.io/docs/query#istartswith)
- [`iendswith`](https://sq.io/docs/query#iendswith)

**Wildcard matchers** (`%` and `_` are wildcards, like SQL `LIKE`):

- [`like`](https://sq.io/docs/query#like)
- [`ilike`](https://sq.io/docs/query#ilike)

Your original example becomes:

```shell
sq '.actor | .first_name, .last_name | where(startswith(.first_name, "TO"))'
sq '.actor | .first_name, .last_name | where(contains(.first_name, "TO"))'
```

See the [Query Guide](https://sq.io/docs/query) for per-driver behavior (there are some SQLite ASCII case-insensitivity quirks worth noting).
